packages/markers/src/browser/problem/problem-decorator.ts

```diff
@@ -38,6 +42,12 @@ export class ProblemDecorator implements TreeDecorator {
         this.problemManager.onDidChangeMarkers(() => this.fireDidChangeDecorations((tree: Tree) => this.collectDecorators(tree)));
     }
 
+    @postConstruct()
+    protected init(): void {
+        this.problemPreferences.onPreferenceChanged(() =>
```

Comment:
Should not we check that `problems.decorations.enabled` was affected? and only then fire an event

Reply:
I initially wanted to do it like the way you mentioned but did not know how to fire the event  using the `tree` which is needed by the emitter. In this way I wouldn't need to refactor the emitter and it'd still work.

Do you see a preferred way forward?

Reply:
not, sure what you mean, I meant:
```ts
this.problemPreferences.onPreferenceChanged(({ preferenceName }) => {
    if (preferenceName === 'problems.decorations.enabled') {
        this.fireDidChangeDecorations((tree: Tree) => this.collectDecorators(tree)));
    }
});
```

Reply:
Thanks, I've updated the code like you mentioned.
I was confused myself I thought perhaps a `handlePreferenceChange` method was needed where I fire the event but the problem was I didn't have a `tree` to pass.